Search hang up with SQ8H

**Describe the bug**
Search no result returned in 30min.
```
2020-03-28 10:14:19,431 | ERROR | default | [ENGINE] SearchTask encounter exception: Error in virtual milvus::knowhere::DatasetPtr milvus::knowhere::IVF::Query(const DatasetPtr&, const Config&) at IndexIVF.cpp:138: Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 2750021632 bytes (error 2 out of memory)
2020-03-28 10:14:21,988 | ERROR | default | [ENGINE] Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 537600000 bytes (error 2 out of memory)
2020-03-28 10:14:23,342 | ERROR | default | [ENGINE] SearchTask encounter exception: Error in virtual milvus::knowhere::DatasetPtr milvus::knowhere::IVF::Query(const DatasetPtr&, const Config&) at IndexIVF.cpp:138: Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 2750021632 bytes (error 2 out of memory)
2020-03-28 10:14:29,078 | ERROR | default | [ENGINE] Error in void faiss::gpu::allocMemorySpaceV(faiss::gpu::MemorySpace, void**, size_t) at gpu/utils/MemorySpace.cpp:26: Error: 'err == cudaSuccess' failed: failed to cudaMalloc 529645568 bytes (error 2 out of memory)
```
```
2020-03-28 10:14:30,911 | DEBUG | default | [KNOWHERE] IVF search cost: 1.67363e+06, quantization cost: 1100.95, data search cost: 572.642
2020-03-28 10:14:30,911 | DEBUG | default | [SERVER] ExecutionEngineImpl::Search float: query done (1.673749 seconds [1673.749019 ms])
2020-03-28 10:14:30,911 | DEBUG | default | [ENGINE] get uids 4137856 from index /test/milvus/db_data_7/sift_1b_2048_128_l2_sq8h_wal/db/tables/sift_1b_2048_128_l2/1583606174108783000/1583760564255358000
2020-03-28 10:14:30,912 | DEBUG | default | [SERVER] ExecutionEngineImpl::Search float: map uids 64000 (0.001357 second [1.357456 ms])
2020-03-28 10:14:30,912 | DEBUG | default | [SERVER] DoSearch file id:16441: job 9 nq 1000 topk 64, do search (1.675195 seconds [1675.194678 ms])
2020-03-28 10:14:30,913 | DEBUG | default | [SERVER] DoSearch file id:16441: job 9 nq 1000 topk 64, reduce topk (0.000412 second [0.412053 ms])
2020-03-28 10:14:30,913 | DEBUG | default | [SERVER] SearchJob 9 finish index file: 16441
2020-03-28 10:14:30,913 | DEBUG | default | [SERVER] DoSearch file id:16441: totally cost (1.675635 seconds [1675.635213 ms])
```
log path:
```
/test/milvus/db_data_7/sift_1b_2048_128_l2_sq8h_wal/logs/milvus-20-03-28-10:05-debug.log
```
**Steps/Code to reproduce behavior**
index_type: ivf_sq8h
dataset: sift_1b_2048_128_l2_sq8h
search.nprobe: 512
config:
```
  collections:
    -
      server:
        db_config.primary_path: /test/milvus/db_data_7/sift_1b_2048_128_l2_sq8h_wal
        cache_config.cpu_cache_capacity: 150
        engine_config.use_blas_threshold: 1100
        engine_config.gpu_search_threshold: 1
        gpu_resource_config.enable: true
        gpu_resource_config.cache_capacity: 4
        gpu_resource_config.search_resources:
          - gpu0
          - gpu1
        gpu_resource_config.build_index_resources:
          - gpu0
          - gpu1
      collection_name: sift_1b_2048_128_l2
      top_ks: [64]
      nqs: [1000]
      search_params:
        nprobe: [1, 2, 4, 8, 16, 32, 64, 128, 256, 512]
```
**note:** search with nprobe=1,2,..256 is OK!

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details**
0.7.1-GPU
run on 2GPU(12G/perf card)

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,6 +21,7 @@ Please mark all change in change log and use the issue from GitHub
 -   \#1735 Fix search out of memory with ivf_flat 
 -   \#1747 Expected error status if search with partition_tag not existed
 -   \#1756 Fix memory exhausted during searching 
+-   \#1781 Fix search hang with SQ8H 
 
 ## Feature
 -   \#1603 BinaryFlat add 2 Metric: Substructure and Superstructure

--- a/core/src/db/engine/ExecutionEngineImpl.cpp
+++ b/core/src/db/engine/ExecutionEngineImpl.cpp
@@ -551,11 +551,11 @@ ExecutionEngineImpl::CopyToGpu(uint64_t device_id, bool hybrid) {
              */
             ENGINE_LOG_DEBUG << "CPU to GPU" << device_id << " start";
             auto gpu_cache_mgr = cache::GpuCacheMgr::GetInstance(device_id);
-            gpu_cache_mgr->Lock();
-            gpu_cache_mgr->Reserve(index_->Size());
+            // gpu_cache_mgr->Lock();
+            // gpu_cache_mgr->Reserve(index_->Size());
             index_ = knowhere::cloner::CopyCpuToGpu(index_, device_id, knowhere::Config());
-            gpu_cache_mgr->InsertItem(location_, std::static_pointer_cast<cache::DataObj>(index_));
-            gpu_cache_mgr->Unlock();
+            // gpu_cache_mgr->InsertItem(location_, std::static_pointer_cast<cache::DataObj>(index_));
+            // gpu_cache_mgr->Unlock();
             ENGINE_LOG_DEBUG << "CPU to GPU" << device_id << " finished";
         } catch (std::exception& e) {
             ENGINE_LOG_ERROR << e.what();

